src/native/libs/System.Native/pal_io.c

```diff
@@ -1949,6 +1949,22 @@ int32_t SystemNative_PWrite(intptr_t fd, void* buffer, int32_t bufferSize, int64
     return (int32_t)count;
 }
 
+static int GetAllowedVectorCount(int32_t vectorCount)
+{
+    int allowedCount = (int)vectorCount;
+
+#if defined(IOV_MAX)
+    if (IOV_MAX < allowedCount)
+    {
+        // We need to respect the limit of items that can be passed in iov.
+        // In case of writes, the managed code is reponsible for handling incomplete writes.
+        // In case of reads, we simply returns the number of bytes read and it's up to the users.
+        allowedCount = IOV_MAX;
```

Comment:
Shouldn't `sysconf(_SC_IOV_MAX)` be preferred here?

Reply:
> Shouldn't `sysconf(_SC_IOV_MAX)` be preferred here?

Great question! I could definitely read it once and store in a static field.

@tmds what is your recommendation?

Reply:
I think it's fine to use `IOV_MAX` here. This means we're deferring binary compatibility to the underlying C library like we're doing as well when compiling with an older glibc version and relying that to work on newer glibc versions.

Side-notes:
- The limit is 1024 on Linux. If you are hitting this limit, probably you application would benefit from using larger buffers.
- The Linux kernel has some optimizations (stackallocs) for <= 8 iovs (`UIO_FASTIOV`).
- POSIX says `IOV_MAX` should be at least 16 (`_POSIX_UIO_MAXIOV`).

Reply:
@tmds thank you!